.github/workflows/pypi-package.yml

```
@@ -0,0 +1,52 @@
+---
+name: Build and upload package
+
+on:
+  push:
+    branches: [main]
+    tags: ["*"]
+  release:
+    types:
+      - published
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  build-package:
+    name: Build & verify package
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: hynek/build-and-inspect-python-package@v2
+
+  auto-release-test-pypi:
+    runs-on: ubuntu-latest
+    needs: [build-package]
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

> Is there a clever way to trigger the workflow for testing without pushing changes to the main branch?

Comment this out for now:

```suggestion
    # if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```